wallet/cardano-wallet.js

```diff
@@ -188,37 +185,38 @@ const CardanoWallet = (secretOrMnemonic, CARDANOLITE_CONFIG) => {
   }
 
   async function getChangeAddress(usedAddressesLimit = Number.MAX_SAFE_INTEGER, offset = 0) {
-    const usedAddressesAndSecrets = await getUsedAddressesAndSecrets()
+    const usedAddressesWithHdNodes = await getUsedAddressesWithHdNodes()
 
     let result
 
-    if (usedAddressesAndSecrets.length < usedAddressesLimit) {
+    if (usedAddressesWithHdNodes.length < usedAddressesLimit) {
       const highestUsedChildIndex = Math.max(
         0x8000000,
-        ...usedAddressesAndSecrets.map((item) => item.childIndex)
+        ...usedAddressesWithHdNodes.map((item) => item.childIndex)
       )
 
-      result = (await getAddressAndSecret(highestUsedChildIndex + 1 + offset)).address
+      result = (await getAddressWithHdNode(highestUsedChildIndex + 1 + offset)).address
```

Comment:
I think we should assert `offset >= 0` to avoid any weird situations when somebody sends us wrong argument

Reply:
fixed a few lines up